cmd/timestamp-cli/app/verify.go

```diff
@@ -130,22 +130,18 @@ func verifyArtifactWithTSR(ts *timestamp.Timestamp) error {
 		return err
 	}
 
-	h := ts.HashAlgorithm.New()
-	b := make([]byte, h.Size())
-
-	r := bytes.NewReader(artifactBytes)
-	n, err := r.Read(b)
-	if err == io.EOF {
-		return err
-	}
+	return verifyHashedMessages(ts.HashAlgorithm.New(), ts.HashedMessage, artifactBytes)
+}
 
-	_, err = h.Write(b[:n])
+func verifyHashedMessages(hashAlg hash.Hash, hashedMessage []byte, artifactBytes []byte) error {
```

Comment:
```suggestion
func verifyHashedMessages(hashAlg hash.Hash, hashedMessage []byte, artifactReader io.Reader) error {
```

While TSRs aren't likely very big, there's no need to copy around []byte slices if you're just taking a pass over it to compute a hash and not doing anything else

Reply:
Sure, done.